drivers/adc/adc_demo/adc_demo.h

```diff
@@ -50,22 +50,39 @@
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
 
-#define MAX_REG_ADDR		16
+#define MAX_ADC_ADDR		16
+#define DEFAULT_LOCAL_SAMPLES 128
 
 /**
  * @struct iio_demo_adc_desc
  * @brief Desciptor.
  */
 struct adc_demo_desc {
 	/** Dummy registers of device for testing */
-	uint8_t reg[MAX_REG_ADDR];
+	uint8_t reg[MAX_ADC_ADDR];
+	/** Demo global device attribute */
+	uint32_t adc_global_attr;
+	/** Demo device channel attribute */
+	uint32_t adc_ch_attr;
```

Comment:
uint32_t adc_ch_attr[TOTAL_CHANNELS]

This way you cand change an attribute for each channel.
And at line 263: desc->adc_ch_attr[channel->ch_num] = value;